Update deploy-code.md (#1458)
astronomer/docs: astro/deploy-code.md

import {siteVariables} from '@site/src/versions';

Using the Astro CLI to push your Astro project, including your DAG code, to a Deployment is the foundation for managing changes on Astro. Astro supports two types of changes:

- Changes to your DAGs. This includes all Python files in the `dags` directory of your Astro project.
- Changes to the base Docker image for your Astro project. This includes your `Dockerfile`, Astro Runtime image version, and all Python and OS-level packages.

> By default, deploys to Astro will push both your DAGs and the rest of the files in your Astro project in a single step. This method is referred to in documentation as image-only deploys. To enable the ability to push only DAGs to Astro for a faster and more reliable development experience, you must enable the [DAG-only deploys](deploy-code.md#DAG-only-deploys) feature.

Reviewer: ```suggestion
By default, deploys to Astro push your DAGs and your Astro project files in a single action. This is an image-only deploy. You can enable [DAG-only deploys](deploy-code.md#DAG-only-deploys) to reduce the deploy time and push only your DAGs to Astro.
```

> To run `astro deploy --dags`, you must first enable the [DAG-only deploys](deploy-code.md#DAG-only-deploys) feature for each Deployment.

Author: ```suggestion
To run `astro deploy --dags`, you must first enable the [DAG-only deploys](deploy-code.md#deploy-dags-only) feature for each Deployment.
```

    ![Docker image information](/img/docs/image-tag-airflow-ui-astro.png)

### What happens during a code deploy

> When you deploy code to Astro, your Astro project is built into a Docker image. This includes system-level dependencies, Python-level dependencies, and your `Dockerfile`. It does not include any of the metadata associated with your local Airflow environment, including task history and Airflow connections or variables that were set locally. This Docker image is then pushed to all containers running Apache Airflow on Astro. If the DAG-only deploy feature is enabled for your Deployment, the `/dags` directory is excluded from this Docker image and pushed separately.

Reviewer: ```suggestion
When you deploy code to Astro, your Astro project is built into a Docker image. This includes system-level dependencies, Python-level dependencies, and your `Dockerfile`. It does not include any of the metadata associated with your local Airflow environment, including task history and Airflow connections or variables that were set locally. This Docker image is then pushed to all containers running Apache Airflow on Astro. If the DAG-only deploy feature is enabled for your Deployment, the `/dags` directory is excluded from this Docker image and is pushed separately.
```

:::

## Deploy DAGs only

:::caution

> This feature is in Public Preview and available to all Astro customers. It is still in development and features and functionality are subject to change.

Reviewer: ```suggestion
This feature is in Public Preview and it is available to all Astro customers. It is still in development, and features and functionality are subject to change.
```

:::

> There are some scenarios where you might want to deploy only the `dags` directory of your Astro project. To do this, enable the DAG-only deploy feature for each Deployment. You must do this only once.

Reviewer: ```suggestion
There are some scenarios where you might want to deploy only the `dags` directory of your Astro project. To do this, you can enable the DAG-only deploy feature for each Deployment. You only need to enable the feature once.
```

:::

To enable the ability to push only DAGs to Astro for a faster development experience, you must enable the [DAG-only deploys](deploy-code.md#DAG-only-deploys) feature once for each Deployment. When it is enabled, you must still run `astro deploy` when you make a change to any file in your Astro project that is not in the `dags` directory.

Enabling DAG-only deploys on Astro has a few benefits:

- DAG-only deploys are significantly faster than running `astro deploy` when you have only made changes to your DAGs.
- When you run `astro deploy --dags`, the workers and schedulers in your Deployment will pick up your changes gracefully and will not restart. This results in a more efficient use of running workers and no downtime for your Deployment.
- You can have different sets of users deploy project changes versus DAG changes. See [DAG-based workflows](ci-cd.md#dag-based-workflows) for how you can set this up in your CI/CD pipelines.

### Enable DAG-only deploys

Enabling DAG-only deploys requires Astro CLI version 1.7 or above. See [Install the Astro CLI](cli/install-cli.md).

> 1. Run:

Reviewer 2: ```suggestion
1. Run the following command:
```

:::

## Deploy DAGs only

:::caution

This feature is in Public Preview and it is available to all Astro customers. It is still in development, and features and functionality are subject to change.

:::

> To enable the ability to push only DAGs to Astro for a faster development experience, you must enable the [DAG-only deploys](deploy-code.md#DAG-only-deploys) feature for each Deployment. You only need to enable the feature once. When it is enabled, you must still run `astro deploy` when you make a change to any file in your Astro project that is not in the `dags` directory.

Author: ```suggestion
To enable the ability to push only DAGs to Astro for a faster development experience, you must enable the feature for each Deployment. You only need to enable the feature once. When it is enabled, you must still run `astro deploy` when you make a change to any file in your Astro project that is not in the `dags` directory.
```
